File: ProgrammersLifeiOS/ProgrammersLifeiOS/Feature.m
//
//  Feature.m
//  ProgrammersLifeiOS
//
//  Created by Zackary T Misso on 7/6/15.
//  Copyright © 2015 Zackary Misso. All rights reserved.
//

#import "Feature.h"

@implementation Feature

-(id)init{
    if(self = [super init]){
        // to be implemented
    }
    return self;
}

// getter methods
-(float)getFeatureDesignScore { return featureDesignScore; }
-(float)getDifficulty { return difficulty; }
-(int)getNumberOfAdditionsNeededAtBest { return numberOfAdditionsNeededAtBest; }
-(int)getNumberOfAdditionsMade { return numberOfAdditionsMade; }
-(int)getNumberOfDeletionsMade { return numberOfDeletionsMade; }

// setter methods
-(void)setFeatureDesignScore:(float)value { featureDesignScore = value; }
-(void)setDifficulty:(float)value { difficulty = value; }
-(void)setNumberOfAdditionsNeededAtBest:(int)value { numberOfAdditionsNeededAtBest = value; }
-(void)setNumberOfAdditionsMade:(int)value { numberOfAdditionsMade = value; }
-(void)setNumberOfDeletuibsMade:(int)value { numberOfDeletionsMade = value; }

@end

File: ProgrammersLifeDemo/src/dataStructures/Feature.java
// This is the Game Tester's version of ProgrammersLife
// Author :: Zackary Misso / FiralStudios
// Version :: 0.0.1

package dataStructures;

public class Feature {
	private float featureDesignScore;
	private float difficulty;
	private int numberOfAdditionsNeededAtBest;
	private int numberOfAdditionsMade;
	private int numberOfDeletionsMade;
	
	// getter methods
	public float getFeatureDesignScore(){return featureDesignScore;}
	public float getDifficulty(){return difficulty;}
	public int getNumberOfAdditionsNeededAtBest(){return numberOfAdditionsNeededAtBest;}
	public int getNumberOfAdditionsMade(){return numberOfAdditionsMade;}
	public int getNumberOfDeletionsMade(){return numberOfDeletionsMade;}
	
	// setter methods
	public void setFeatureDesignScore(float param){featureDesignScore = param;}
	public void setDifficulty(float param){difficulty = param;}
	public void setNumberOfAdditionsNeededAtBest(int param){numberOfAdditionsNeededAtBest = param;}
	public void setNumberOfAdditionsMade(int param){numberOfAdditionsMade = param;}
	public void setNumberOfDeletionsMade(int param){numberOfDeletionsMade = param;}
}
